File: C++/API/Treehopper.Libraries/src/Sensors/Pressure/PressureSensor.cpp
#include "Sensors/Pressure/PressureSensor.h"

namespace Treehopper
{
	namespace Libraries
	{
		namespace Sensors
		{
			namespace Pressure
			{
				double PressureSensor::pascal()
				{
					if (autoUpdateWhenPropertyRead) update();

					return _pascal;
				}

				double PressureSensor::bar()
				{
					return pascal() / 100000.0;
				}

				double PressureSensor::atm()
				{
					return pascal() / (1.01325 * 100000.0);
				}

				double PressureSensor::psi()
				{
					return atm() / 14.7;
				}
			}
		}
	}
}

File: Java/api/treehopper.libraries/src/main/java/io/treehopper/libraries/sensors/pressure/PressureSensor.java
package io.treehopper.libraries.sensors.pressure;

import io.treehopper.libraries.sensors.IPollable;

public abstract class PressureSensor implements IPollable {

    protected boolean autoUpdateWhenPropertyRead;
    protected double pascal = 0;


    @Override
    public boolean isAutoUpdateWhenPropertyRead() {
        return autoUpdateWhenPropertyRead;
    }

    @Override
    public void setAutoUpdateWhenPropertyRead(boolean value) {
        autoUpdateWhenPropertyRead = value;
    }

    public double getPascal() {
        if (isAutoUpdateWhenPropertyRead())
            update();

        return pascal;
    }

    public double getBar() {
        return getPascal() / 100000.0;
    }

    public double getAtm() {
        return getPascal() / (1.01325 * 100000.0);
    }

    public double getPsi() {
        return getAtm() / 14.7;
    }
}
